src/util/promiseRetry.ts

```diff
@@ -0,0 +1,32 @@
+type PromiseGenerator = () => Promise<any>;
+
+const DEFAULT_RETRY_ATTEMPTS = 3;
+const DEFAULT_RETRY_DELAY = 100;
+
+const wait = (delay: number): Promise<undefined> =>
+  new Promise((resolve): number => setTimeout(resolve, delay));
+
+const retry = (
+  func: PromiseGenerator,
+  times: number = DEFAULT_RETRY_ATTEMPTS,
+  delay: number = DEFAULT_RETRY_DELAY
+): Promise<any> =>
+  new Promise(
+    (resolve, reject): void => {
+      func()
```

Comment:
could this be `async`?

Reply:
It could, but sadly we can't afford the overhead of `async/await` transpilation in this library.

Babel transform async functions using the [regenerator/runtime](https://github.com/facebook/regenerator/tree/master/packages/regenerator-runtime), i.e. it compiles them to generator functions and then includes a runtime polyfill into your bundle via [`@babel/plugin-transform-regenerator`](https://babeljs.io/docs/en/babel-plugin-transform-regenerator). The runtime is around ~6-8kb in size, so not worth it just for syntactic sugar.

Reply:
According to caniuse.com async is pretty close to same support as `Promise`
https://caniuse.com/#search=await
What's the thought about raw `async` no transpilation?

Reply:
> What's the thought about raw async no transpilation?

I/we need to do the research to see how many browsers we'd be losing if we went down this route and also need to run a separate round of tests to see if WebWorker support for async is the same as the `window` (which historically isn't the case). So I still want to keep this util until we have that discussion as a group.